hibernate6-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/graal/HibernateSubstitutions.java

```diff
@@ -0,0 +1,242 @@
+ * limitations under the License.
+ */
+package io.micronaut.configuration.hibernate.jpa.graal;
+
+/*
+ * Internal class that provides substitutions for Hibernate on GraalVM substrate.
+ *
+ * @author graemerocher
+ * @since 1.2.0
+ */
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import io.micronaut.configuration.hibernate.jpa.proxy.IntrospectedHibernateBytecodeProvider;
+import io.micronaut.core.annotation.TypeHint;
+import io.micronaut.core.annotation.TypeHint.AccessType;
+import io.micronaut.jdbc.spring.HibernatePresenceCondition;
+import org.hibernate.boot.archive.spi.InputStreamAccess;
+import org.hibernate.boot.jaxb.internal.MappingBinder;
+import org.hibernate.boot.jaxb.spi.Binding;
+import org.hibernate.boot.model.naming.ImplicitNamingStrategyJpaCompliantImpl;
+import org.hibernate.boot.registry.classloading.spi.ClassLoaderService;
+import org.hibernate.bytecode.spi.BytecodeProvider;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.id.Assigned;
+import org.hibernate.id.ForeignGenerator;
+import org.hibernate.id.GUIDGenerator;
+import org.hibernate.id.IdentityGenerator;
+import org.hibernate.id.IncrementGenerator;
+import org.hibernate.id.SelectGenerator;
+import org.hibernate.id.UUIDGenerator;
+import org.hibernate.id.UUIDHexGenerator;
+import org.hibernate.id.enhanced.HiLoOptimizer;
+import org.hibernate.id.enhanced.LegacyHiLoAlgorithmOptimizer;
+import org.hibernate.id.enhanced.NoopOptimizer;
+import org.hibernate.id.enhanced.PooledLoOptimizer;
+import org.hibernate.id.enhanced.PooledLoThreadLocalOptimizer;
+import org.hibernate.id.enhanced.PooledOptimizer;
+import org.hibernate.id.enhanced.SequenceStyleGenerator;
+import org.hibernate.id.enhanced.TableGenerator;
+import org.hibernate.persister.collection.BasicCollectionPersister;
+import org.hibernate.persister.collection.OneToManyPersister;
+import org.hibernate.persister.entity.SingleTableEntityPersister;
+import org.hibernate.service.ServiceRegistry;
+import org.hibernate.tuple.entity.EntityMetamodel;
+import org.hibernate.type.EnumType;
+
+import javax.xml.stream.XMLResolver;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Properties;
+
+// Additional classes
+@TypeHint(
+        typeNames = {
+                "org.hibernate.internal.CoreMessageLogger_$logger",
+                "org.hibernate.internal.EntityManagerMessageLogger_$logger",
+                "org.hibernate.annotations.common.util.impl.Log_$logger",
+                "com.sun.xml.internal.stream.events.XMLEventFactoryImpl",
+                "org.hibernate.bytecode.enhance.spi.interceptor.BytecodeInterceptorLogging_$logger",
+                "org.hibernate.envers.internal.EnversMessageLogger_$logger",
+        }
+)
+final class Loggers {
+}
+
+// For HQL and other Hibernate internal classes
+@TypeHint(value = {
```

Comment:
It would be nice to check if those are still needed

Reply:
Not sure, probably are. Just removed those that are not present in hibernate6